DeusEx/Classes/DXRKeys.uc

```diff
@@ -137,46 +137,138 @@ function bool KeyPositionGood(NanoKey k, vector newpos)
     return True;
 }
 
-function AdjustRestrictions(int doorspickable, int doorsdestructible, int deviceshackable, int removeinvisiblewalls)
+function AdjustRestrictions(int doorsmode, int doorspickable, int doorsdestructible, int deviceshackable, int removeinvisiblewalls)
 {
-    local DeusExMover d;
     local Keypoint kp;
 
+    switch( (doorsmode/256) ) {
```

Comment:
maybe this is trying to be too clever, I could make a couple of static helper functions to make it easier to read and less error prone